elastic/kibana
# [Discover] Context - Fix bug when document id contains a slash

## Summary

In the Elastic Stack 7.0 types were deprecated, in [8.0 they are removed](https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html#_schedule_for_removal_of_mapping_types).  In Discover's context URL types have been a part until 7.4. With 7.5 and #44008 they were removed, and a redirect of the legacy to the new URL was implemented. This redirect broke the context view for URLs with document ids containing slashes `/`. This PR removes this legacy redirect, moving the redirect of legacy URLs to a "higher level" using our "url_forwarding" service.

### How to test

1. Create some record in Elasticsearch 

```
POST _bulk
{ "index" : { "_index" : "context-special","_id" : "TLmBgdeL0GOgAqCX+Q1JSRE6Ijlqg95S71GIGS/JYkc=2" } }
{ "timestamp":"2020-05-05T20:00:00.123Z", "name": "test" }

POST _bulk
{ "index" : { "_index" : "context-special","_id" : "TLmBgdeL0GOgAqCX+Q1JSRE6Ijlqg95S71ddsGIGS/JYkc=2" } }
{ "timestamp":"2020-05-05T21:00:00.123Z", "name": "test" }
```

2. Create an index pattern in stack management 

3. Navigate to discover, adapt time range to view those documents

4. Unfold a document and select 'View surrounding documents'

Before this PR this should not work, with this PR is works again 

Fixes #77443
